[Ria Khaitan] iP

riakhaitan opened the pull request:
# Halloumi ChatBot ^_^

> *"Your mind is for having ideas, not holding them." -David Allen*

The Halloumi ChatBot keeps track of your tasks and progress, *all in one place*. It's,
* ~~easy~~ **SUPER EASY** to use ⏩
* text-based 📝
* offline 🌳

If you're a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

All you need to do is,
1. Sign up [here](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#links) for **free**
2. Complete the demo
3. Start adding your tasks

Features:
- [x] Task management
- [ ] Managing deadlines *(coming soon)*
- [ ] Reminders *(coming soon)*

Some commands to get you started with Halloumi are:
```
list
todo <insert task here>
mark 1
```

Commit:
Add Gradle support

riakhaitan committed:
add level 1

riakhaitan committed:
update gitignore

riakhaitan committed:
Add, List

riakhaitan committed:
mark as done

riakhaitan committed:
todos, events, deadlines

riakhaitan committed:
add UI testing

riakhaitan committed:
handle errors

riakhaitan committed:
add delete

riakhaitan committed:
add level 7 to branch-level-7

rahulprasad01 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,108 @@
+import exceptions.DukeExceptions;
```
Should the import statements be ordered differently?

rahulprasad01 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,108 @@
+        System.out.println("|-|-|-|-|-|-|-|-|-|-|-|-|-|-|-|-|-|-|-|-|-|-|-|-|-|-|-|-|-|-|-|-|-|-|-|");
+    }
+
+    public static String makeDesc(String[] text, int len) {
```
Perhaps a more intuitive variable name here?

rahulprasad01 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,58 @@
+    }
+}
+
+class ToDo extends Task {
```
I like how the code has been indented

rahulprasad01 reviewed (commented):
Overall really clean code

riakhaitan committed:
dates and times

riakhaitan committed:
fix merge conflicts{

markbiju commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,209 @@
+            File f = new File("/Users/riakhaitan/iP/ip/data/duke.txt");
+            f.createNewFile();
+
+            boolean directory = f.getParentFile().mkdirs();
```
Just a suggestion, perhaps you could use a name for this variable that is more boolean-sounding such as isDirectory? Good use of mkdir() here, creating a directory to store data

markbiju commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,62 @@
+        when = date;
+    }
+    public String toString() {
+        return "[D]" + super.toString() + " (: " +
```
A good practice would be to place a line break before an operator! Good use of line break to improve readability

markbiju commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,209 @@
+            String[] textSplit = textSplitOne[0].split(" ");
+            String fullDesc = makeDesc(textSplit, textSplit.length);
+            try {
+                switch (textSplit[0]) {
```
For case clauses, there is no indentation

markbiju commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,209 @@
+                        System.out.println("You have " + lists.size() + " tasks left now! ^_^");
+                        lineTwo();
+                        break;
+                    case "deadline":
```
Logical units within a block should be separated by a line, perhaps a space after each case would be ideal!

markbiju reviewed (commented):
A good effort at following Java coding standards, with appropriate method naming and use of camelCase!

riakhaitan committed:
more OOP

riakhaitan committed:
java doc

riakhaitan committed:
packages

riakhaitan committed:
coding standard

riakhaitan committed:
find

riakhaitan committed:
add junit tests

riakhaitan committed:
create a JAR file

riakhaitan committed:
add gradle support
Merge remote-tracking branch 'origin/add-gradle-support' into Add-gradle-support

riakhaitan committed:
add gradle support

riakhaitan committed:
java doc

riakhaitan committed:
coding standard

riakhaitan committed:
fix find

riakhaitan committed:
fix coding standard
Merge branch 'branch-A-CodingStandard'

riakhaitan committed:
merge find

riakhaitan committed:
use assertions

riakhaitan committed:
code quality

riakhaitan committed:
Merge pull request #2 from riakhaitan/branch-A-Assertions

use assertions

riakhaitan committed:
Merge pull request #3 from riakhaitan/branch-A-CodeQuality

code quality

riakhaitan committed:
Set theme jekyll-theme-cayman

riakhaitan committed:
Update ReadMe

riakhaitan committed:
build.gradle : Update dependancies in the file to add JavaFX dependencies

To implement JavaFX and incorporate gui in our code, we need to add the related dependencies for it.

We add the corresponding dependencies to our build.gradle so we are able to use JavaFX and incorporate gui in our code.

We need to add dependencies to our build.gradle so we can use gradle to build our gui, this is the most efficient way.

riakhaitan committed:
add bcd extension

riakhaitan committed:
fix code for event

riakhaitan committed:
Update README.md

riakhaitan committed:
add javafx

riakhaitan committed:
Merge branch 'master' of https://github.com/riakhaitan/ip

riakhaitan committed:
add ui.png
